components/match2/wikis/warcraft/match_legacy.lua

```diff
@@ -0,0 +1,281 @@
+
+local UNKNOWNREASON_DEFAULT_LOSS = 'L'
+local TBD = 'TBD'
+
+---@param match table
+---@param options {storeMatch2:boolean, storeMatch1:boolean, storePageVar:boolean, bracketId: string}
+---@return nil
+function MatchLegacy.storeMatch(match, options)
+	if not options.storeMatch1 or #match.match2opponents ~= 2 or not Namespace.isMain() then return end
+
+	match = MatchLegacy._parseJsons(match)
+
+	if Array.all(match.match2opponents, function(opponent) return opponent.type == Opponent.solo end) then
+		return MatchLegacy._storeSoloMatch(match)
+	end
+
+	if Array.all(match.match2opponents, function(opponent) return opponent.type == Opponent.team
+		or opponent.type == Opponent.solo end) then return MatchLegacy._storeTeamMatch(match) end
+end
+
+---@param object table
+---@return table
+function MatchLegacy._parseJsons(object)
+	for key, item in pairs(object) do
+		object[key] = Json.parseIfTable(item) or item
+		if type(object[key]) == 'table' then
+			object[key] = MatchLegacy._parseJsons(object[key])
+		end
+	end
+
+	return object
+end
+
+---@param match2 table
+function MatchLegacy._storeSoloMatch(match2)
+	local match = MatchLegacy._convertParameters(match2)
+	match.mode = '1v1'
+	match.objectName = 'legacymatch_' .. match2.match2id
+
+	mw.ext.LiquipediaDB.lpdb_match(match.objectName, match)
+
+	match.extradata = Json.parseIfString(match.extradata)
+
+	for gameIndex, game in ipairs(match2.match2games or {}) do
+		MatchLegacy._storeGame(game, gameIndex, match)
+	end
+end
+
+---@param match2 table
+function MatchLegacy._storeTeamMatch(match2)
+	local match = MatchLegacy._convertParameters(match2)
+	match.mode = Array.all(match2.match2opponents, function(opponent)
+		return opponent.type == Opponent.team end) and Opponent.team or 'mixed'
+	match.objectName = 'legacymatch_' .. match2.match2id
+
+	mw.ext.LiquipediaDB.lpdb_match(match.objectName, match)
+
+	match.extradata = Json.parseIfString(match.extradata)
+
+	for sumbmatchIndex, submatch in Table.iter.spairs(MatchLegacy._groupIntoSubmatches(match2, match.objectName)) do
+		MatchLegacy._storeSubMatch(submatch, sumbmatchIndex, match)
+	end
+end
+
+---@param match2 table
+---@return table
+function MatchLegacy._convertParameters(match2)
+	local bracketData = match2.match2bracketdata or {}
+
+	local match = Table.filterByKey(match2, function(key, entry) return not String.startsWith(key, 'match2') end)
+	---@cast match table
+	match.type = match.type and string.lower(match.type) or nil
+	match.header = DisplayHelper.expandHeader(Logic.emptyOr(bracketData.header, bracketData.inheritedheader, ''))[1]
+	match.tickername = Variables.varDefault('tournament_tickername', Variables.varDefault('tournament_ticker_name'))
+
+	match.extradata = match.extradata or {}
+	local extradata = match.extradata --[[@as table]]
+
+	match.staticid = (tonumber(Variables.varDefault('matchID')) or 0) + 1
+	Variables.varDefine('matchID', match.staticid)
+	extradata.legacymatchid = match.staticid
```

Comment:
Why is legacymatchid needed?

Reply:
to not break api v1/match consumers
they need it for the old match1 matches for some reason and to not break their data usage ...

Reply:
Don't see any consumers using it

![image](https://github.com/Liquipedia/Lua-Modules/assets/3426850/63199d26-da53-4ded-bbea-7341636def6e)

Reply:
yeah external